packages/backend/server/src/base/utils/request.ts

```diff
@@ -118,3 +118,29 @@ export function getRequestIdFromHost(host: ArgumentsHost) {
   const req = getRequestFromHost(host);
   return getRequestIdFromRequest(req, type);
 }
+
+const AFFINE_VERSION_REGEX =
+  /AFFiNE(?:-\w{1,50})?\/(\d{1,50}\.\d{1,50}\.\d{1,50}(?:-\w{1,50}\.\d{1,50})?)/i;
+
+export function getClientVersionFromRequest(req: Request) {
+  let version = req.headers['x-affine-version'];
+  if (Array.isArray(version)) {
+    version = version[0];
+  }
+  if (version) {
+    return version;
+  }
+
+  // fallback to `user-agent`
+  const userAgent = req.headers['user-agent'];
+  if (userAgent) {
+    // AFFiNE/0.22.2
+    // AFFiNE-beta/0.23.0-beta.2
+    const match = AFFINE_VERSION_REGEX.exec(userAgent);
+    if (match) {
+      return match[1];
+    }
+  }
```

Comment:
not required, client that without affine-version is banned

Reply:
I tried it, and both 0.18 and 0.19 can make requests, and I haven't been banned yet.